src/auditing.md

```diff
@@ -169,9 +169,12 @@ section that follows.
 Example *audit.log* denied and granted events are shown in the following
 examples:
 
+<br>
```

Comment:
Is the "<br>" strictly necessary?   My guess is no, but I haven't generated the HTML or PDF to verify.

I'm asking because it would be nice if the markdown sources were as close to pure markdown as possible, adding HTML tags is something we should try to avoid moving forward.

Reply:
<details>
 <summary><b>with:</b></summary>
<img src="https://user-images.githubusercontent.com/16723362/88162774-16aaee80-cc12-11ea-99aa-b421c2cd0e06.png"/>
</details>

<details>
  <summary><b>without:</b></summary>
<img src="https://user-images.githubusercontent.com/16723362/88162997-54a81280-cc12-11ea-9bf8-638260aa93dd.png"/>
</details>

They add a bit of spacing that makes it a bit more readable, there doesn't seem to be a way to do this in pure markdown (https://stackoverflow.com/questions/20543454/create-two-blank-lines-in-markdown/20543835#20543835)

Reply:
Thanks for the pictures of the rendered results, that really helps.

I still think I would prefer to keep the markdown as clean as possible, preferring to use CSS when necessary to fix or improve the formatting.  Take a look at the "src/styles_{pdf,html}.css" files for our current CSS.

Reply:
I don't really think this is possible to do with CSS, take these examples that currently use CSS:

```html
<p>Notes:</p>
<ol type="1">
<li>It is not mandatory for SELinux-aware applications to audit events or even log them in the audit log. The decision is made by the application designer.</li>
<li>The format of audit messages do not need to conform to any format, however where possible applications should use the <em><strong>audit_log_user_avc_message</strong>(3)</em> function with a suitably formatted message if using <em><strong>auditd</strong>(8)</em>. The type of audit events possible are defined in the <em>include/libaudit.h</em> and <em>include/linux/audit.h</em> files.</li>
<li>Those libselinux library functions that output messages do so to <code>stderr</code> by default, however this can be changed by calling <em><strong>selinux_set_callback</strong>(3)</em> and specifying an alternative log handler.</li>
</ol>
<!-- here we want additional space -->
<p><br></p>
<h2 id="avc-audit-events">AVC Audit Events</h2>
````
This could probably done by adding additional padding above headers, but might also have other unwanted effects.

```html
<p>These are example X-Windows object manager audit message:</p>
<pre><code>type=USER_AVC msg=audit(1267534171.023:18): user pid=1169 uid=0
auid=4294967295 ses=4294967295
subj=system_u:unconfined_r:unconfined_t msg='avc: denied { getfocus }
for request=X11:GetInputFocus comm=X-setest xdevice="Virtual core
keyboard" scontext=unconfined_u:unconfined_r:x_select_paste_t
tcontext=system_u:unconfined_r:unconfined_t tclass=x_keyboard :
exe="/usr/bin/Xorg" sauid=0 hostname=? addr=? terminal=?'

type=USER_AVC msg=audit(1267534395.930:19): user pid=1169 uid=0
auid=4294967295 ses=4294967295
subj=system_u:unconfined_r:unconfined_t msg='avc: denied { read } for
request=SELinux:SELinuxGetClientContext comm=X-setest resid=3c00001
restype=&amp;lt;unknown&amp;gt;
scontext=unconfined_u:unconfined_r:x_select_paste_t
tcontext=unconfined_u:unconfined_r:unconfined_t tclass=x_resource :
exe="/usr/bin/Xorg" sauid=0 hostname=? addr=? terminal=?'</code></pre>
<!-- here we want additional space -->
<p><br></p>
<p>This is an example <strong>granted</strong> audit message:</p>
```

I don't see any way of adding additional spacing using CSS without messing up the entire document (Basically introducing extra space after every paragraph `<p>...</p>`)

Reply:
I haven't posted a patchset or a PR yet (I probably should, even if it is just a WIP/draft PR), but I'm currently in the process of finishing the markdown conversion that Richard started (mostly it is just table conversions at this point).  Originally I started doing this to increase the readability in plain text format as HTML tables are not easily read in source form, unlike Markdown's tables.  However, in the process of doing this I realized that a lot of CSS hacks which were needed to make the PDF render okay-ish we no longer needed.  The lesson learned is that once the source material follows one format, and not a hybrid markdown/HTML format, pandoc does a *much* better job rendering the source into different formats.

I *really* *really* *really* want to see us get the notebook source material into Markdown format wherever it is possible.  Not only does this improve readability in plain text, but it helps separate the content from the formatting giving us more flexibility when rendering different output formats.  I will agree that more should be done in terms of the HTML and PDF CSS formatting, but we need to get everything into Markdown first.

I realize this is all a bit much for a simple line-break tag, but I hope this explains why I'm so adamant on removing raw HTML from the sources.

Reply:
I lost some this weekend when I realized that GH's markdown flavor doesn't support `grid_tables`, but here is the current work-in-progress for moving the source material fully to markdown:
* https://github.com/pcmoore/misc-selinux_notebook/tree/working-md_full

I should create a draft PR for that, I'll do that now.

Reply:
> I realize this is all a bit much for a simple line-break tag, but I hope this explains why I'm so adamant on removing raw HTML from the sources.

I know, I've removed all `<br>` elements from the file.